### PR TITLE
[issue 79] test run on android

### PR DESCRIPTION
**Pull request by andkononykhin**
- restructures the project to follow gradle's [multi sub-projects / modules layout](https://docs.gradle.org/current/userguide/multi_project_builds.html#sec:adding_subprojects)
- adds android module to run java unit tests on an emulator (as instrumented tests)

addresses https://github.com/sicpa-dlab/didcomm-gemini/issues/79

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+dependencies {
+    // project dependnecies
+    androidTestImplementation "com.nimbusds:nimbus-jose-jwt:${nimbusJoseJWTVersion}"
+    androidTestImplementation "com.google.crypto.tink:tink:${googleTinkVersion}"
+    // project itself
+    androidTestImplementation project(':lib')
+
+    // test dependencies
+    androidTestImplementation "org.jetbrains.kotlin:kotlin-test"
+    androidTestImplementation "com.fasterxml.jackson.core:jackson-databind:2.11.1"
```
Comment (asmarty) [inline android/build.gradle:52]: can you move all versions to property file?

File: android/build.gradle [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        androidTest {
+            kotlin.srcDirs += "../lib/src/test/kotlin"
+        }
+    }
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
```
Comment (asmarty) [inline android/build.gradle:37]: very miner thing, can you add a blank space above?

File: build.gradle [hunk 3]
```diff
@@ -1,35 +1,22 @@
...
-
-repositories {
-    mavenCentral()
-    flatDir {
-        // TODO remove when contribution will be merged to upstream
-        dir "lib"
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+buildscript {
+    repositories {
+        google()
```
Comment (asmarty) [inline build.gradle:4]: Are it needed? I thought you move repositories to settings.gradle, please correct me If i'm not right.

File: android/build.gradle (hunk 1 above)
Comment (andkononykhin) [reply android/build.gradle]: done

File: android/build.gradle (hunk 2 above)
Comment (andkononykhin) [reply android/build.gradle:38]: done

File: build.gradle (hunk 3 above)
Comment (andkononykhin) [reply build.gradle:4]: can't answer for sure here, but I thought the same and it was wrong: by some reason it fails if the repos are not listed here as well, I suggest to dig into that later one more time

**Review by asmarty — APPROVED** [review]

**Pull request merged**